File: rust/9.回文数.rs
/*
 * @lc app=leetcode.cn id=9 lang=rust
 *
 * [9] 回文数
 */

// @lc code=start
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x >= 0 && x < 10 {
            return true;
        }
        if x < 0 || x % 10 == 0 {
            return false;
        }

        let mut x = x;
        let mut r = 0;
        while x > r {
            r = r * 10 + x % 10;
            x = x / 10;
        }
        return x == r || x == r / 10;
    }
}
// @lc code=end

File: 9.回文数.go
/*
 * @lc app=leetcode.cn id=9 lang=golang
 *
 * [9] 回文数
 */
package leetcode

// @lc code=start
func isPalindromeNum(x int) bool {
	if x < 0 || (x%10 == 0 && x != 0) {
		return false
	}
	var rN int
	for x > rN {
		rN = rN*10 + x%10
		x /= 10
	}
	return x == rN || x == rN/10
}

// @lc code=end
